### PR TITLE
refactor(carousel): change to use centralized active state

### PR DESCRIPTION
**Pull request by wesleycho**
- Change so active state is primarily on two-way bound variable

BREAKING CHANGE: This changes the active state to be in line with
similar components centralizing the information. Please refer to the
carousel documentation for usage.

This implements #5353.

### CODE REVIEW

**Comment by Foxandxss** [comment]
Needs the doc update.

**Comment by wesleycho** [comment]
Updated with docs

File: src/carousel/docs/README.md [hunk 1]
```diff
@@ -6,53 +6,52 @@ Use a `<uib-carousel>` element with `<uib-slide>` elements inside it.
 
 ### uib-carousel settings
 
+* `active`
+  <i class="glyphicon glyphicon-eye-open"></i>
+  _(Default: Index of first slide)_ -
```
Comment (Foxandxss) [inline src/carousel/docs/README.md]: backsticks

**Comment by Foxandxss** [comment]
LGTM

**Pull request closed without merging**